### PR TITLE
generateMatchedCohort

### PR DESCRIPTION
**Pull request by martaalcalde**
Created the function: generateMatchedCohort

### CODE REVIEW

File: R/exactMatchingCohort.R [hunk 1]
```diff
@@ -1,41 +1,32 @@
...
-  # if(!isTRUE(cases_check)){
-  #   errorMessage$push(glue::glue("- {cdm} has not table named 'cases'"))
-  # }
-  #
-  # checkmate::reportAssertions(collection = errorMessage)
-#  -----------------------------------------------------------------------------
-
-    matchCols <- c()
+exactMatchingCohort <- function(cdm,
+                                name = "NewCohort",
```
Comment (catalamarti) [inline R/exactMatchingCohort.R]: not default

File: R/exactMatchingCohort.R [hunk 2]
```diff
@@ -1,41 +1,32 @@
...
+                                targetCohortName,
+                                targetCohortId = NULL,
+                                matchSex  = TRUE,
+                                matchYear = TRUE,
```
Comment (catalamarti) [inline R/exactMatchingCohort.R]: matchYearOfBirth?

**Review by catalamarti — CHANGES_REQUESTED** [review]

File: R/exactMatchingCohort.R [hunk 3]
```diff
@@ -88,14 +91,45 @@ exactMatchingCohort <- function(cohort, matchSex = TRUE, matchYearBirth = TRUE,
...
+    dplyr::rename("subject_id" = "cases_id") %>%
+    dplyr::union_all(
+      matches %>%
+        dplyr::select(-.data$cases_id) %>%
+        dplyr::rename("subject_id" = "controls_id") %>%
+        dplyr::mutate(cohort_definition_id = .data$cohort_definition_id + 0.5)
     ) %>%
-    dplyr::select("cases_id", "controls_id", dplyr::any_of(matchCols))
-    return(matches)
+    CDMConnector::compute_query()
```
Comment (catalamarti) [inline R/exactMatchingCohort.R]: computeQuery(temporary = FALSE, schema = attr(cdm, "write_schema"), name = name)

File: R/exactMatchingCohort.R [hunk 4]
```diff
@@ -88,14 +91,45 @@ exactMatchingCohort <- function(cohort, matchSex = TRUE, matchYearBirth = TRUE,
...
+
+  cohort_set_ref <- cohort_set_ref %>%
```
Comment (catalamarti) [inline R/exactMatchingCohort.R:119]: cohortSet:
- cohort_definition_id
- cohort_name
- target_cohort_name # tots igual
- target_cohort_id
- mach_status: target/matched
- match_sex # tots igual TRUE/FALSE
- match_year_of_birth # tots igual TRUE/FALSE

**Review by catalamarti — CHANGES_REQUESTED** [review]

File: R/exactMatchingCohort.R (hunk 4 above)
Comment (martaalcalde) [reply R/exactMatchingCohort.R:119]: what do you mean with "target_cohort_id"?

File: R/generateMatchedCohort.R [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#' @param matchYearOfBirth can be TRUE or FALSE
+#' @param ratio at the moment, it is not used this argument
+#'
+#' @return plot
+#' @export
+#'
+#' @examples
+
+generateMatchedCohort <- function(cdm,
+                                  name,
```
Comment (catalamarti) [inline R/generateMatchedCohort.R:17]: name should be snake_case

**Review by catalamarti — CHANGES_REQUESTED** [review]

**Pull request merged**